### PR TITLE
Performance property

### PR DESCRIPTION
**Pull request by bytedissident**
**What does this do?** 
It adds a property called **testPerformance*** to Behave. The intention is multi-fold. 

1. Make it simple to turn on and off performance testing for the developer
2. Add performance testing to the entire suite as opposed to forcing the developer to add it in spots.
3. Associate the events being tested with the slow frame renders in order to help the developer more easily debug their code.
4. Adds a new callback to the run() method, **warn**. The warn callback returns a list of events that were being tested when slow frame renderings occurred.
5. Updates ReadMe to include the new functionality.

By default performance testing is off

*note:* I added cleaned up the ReadMe a bit as well as adding info about this feature.

**To Test**

- Run PerformanceBDDTests
- There are three tests: efficient, inefficient with property on,  inefficient with property off. 
- Examine the console post test run and you will see in the **warn** completion handler a list of events returned as a string that are ties to the slow frame rendering on *inefficient with property on* 
- You should see no warnings on the other two. 
- Also, on the test with "property off" it will have slow renders, but will only fail if the warning is triggered. It should not as the property is set to false in the test .

### CODE REVIEW

File: Example/ExampleAppTests/PerformanceBDDTests.swift [hunk 1]
```diff
@@ -15,34 +15,29 @@ class PerformanceBDDTests: XCTestCase {
...
     // THIS TESTS AN EFFICIENT TABLE SO IT SHOULD PASS
-    func testPerformance_isEfficient() {
+    func testPerformance_isEfficientWithPropertyOn() {
         let expectations = expectation(description: "\(#function)")
         let api = Behaviour()
         navgateToThePerformanceScreen(api)
         api.listen(for: performanceView) {
-            api.measurePerformance()
             let indexPath = IndexPath(row: 5, section: 0)
             api.scrollTableTo(indexPath: indexPath, identfier: self.performanceView)
```
Comment (Vadimkomis) [inline Example/ExampleAppTests/PerformanceBDDTests.swift:24]: Typo. Let's fix it.

File: Example/ExampleAppTests/PerformanceBDDTests.swift [hunk 2]
```diff
@@ -15,34 +15,29 @@ class PerformanceBDDTests: XCTestCase {
     private let efficiencyButton = "efficiency-button"
     
     // THIS TESTS AN EFFICIENT TABLE SO IT SHOULD PASS
-    func testPerformance_isEfficient() {
+    func testPerformance_isEfficientWithPropertyOn() {
         let expectations = expectation(description: "\(#function)")
         let api = Behaviour()
         navgateToThePerformanceScreen(api)
```
Comment (Vadimkomis) [inline Example/ExampleAppTests/PerformanceBDDTests.swift:21]: Spelling mistake. Let's fix it.

File: README.md [hunk 3]
```diff
@@ -45,11 +52,18 @@ api.listen(for: "my-view") {
 3. When the event is triggered the completion handler will be called. Each event added to the test gets tested in the order it was added, **FIFO**. The events are triggered synchronously.
 
 **Run:**
-To make your test run use the run method. Behave tests will not run without explicitly calling run. Run has a *fail* completion handler, this will get triggered if any of your events are not triggered. Behave will pass back the identifier in question so you can identify the issue.
+To make your test run use the run method. Behave tests will not run without explicitly calling run. Run has: **success**, **fail** and **warn** completion handlers.
```
Comment (Vadimkomis) [inline README.md:55]: One suggestion about the readme, is to make it easier for navigation. For example, I like how `Alamofire` has: https://github.com/Alamofire/Alamofire/blob/master/README.md

File: Example/ExampleAppTests/PerformanceBDDTests.swift [hunk 4]
```diff
@@ -53,12 +48,39 @@ class PerformanceBDDTests: XCTestCase {
...
+        api.run(fail: { error in
             XCTFail(error)
+        },warn: { warnings in
+            print(warnings)
+        })
+        waitForExpectations(timeout: api.testTimeInterval)
+    }
+    
+    // THIS TESTS AN INEFFICIENT TABLE BUT PERFORMANCE TESTING IS OFF SO IT SHOULD ONLY FAIL IF THE PROPERTY IS BROKEN
+    func testPerformance_isInEfficientWithPropertyOff() {
```
Comment (Vadimkomis) [inline Example/ExampleAppTests/PerformanceBDDTests.swift:62]: Can you clarify what do you mean by "property is broken" ?

**Review by Vadimkomis — CHANGES_REQUESTED** [review]

File: Example/ExampleAppTests/PerformanceBDDTests.swift (hunk 2 above)
Comment (bytedissident) [reply Example/ExampleAppTests/PerformanceBDDTests.swift:21]: what's the mistake?

Comment (bytedissident) [reply Example/ExampleAppTests/PerformanceBDDTests.swift:21]: 👍

File: Example/ExampleAppTests/PerformanceBDDTests.swift (hunk 1 above)
Comment (bytedissident) [reply Example/ExampleAppTests/PerformanceBDDTests.swift:24]: What's the typo?

File: README.md (hunk 3 above)
Comment (bytedissident) [reply README.md:55]: I like that too, lets do it as a separate PR so I can think about it. I'll add an issue.

Comment (bytedissident) [reply README.md:55]: https://github.com/Freshly/Behave/issues/41

File: Example/ExampleAppTests/PerformanceBDDTests.swift (hunk 4 above)
Comment (bytedissident) [reply Example/ExampleAppTests/PerformanceBDDTests.swift:62]: Updated, changed language.

File: Example/ExampleAppTests/PerformanceBDDTests.swift (hunk 2 above)
Comment (Vadimkomis) [reply Example/ExampleAppTests/PerformanceBDDTests.swift:21]: `navgateToThePerformanceScreen` should be `navigateToThePerformanceScreen`

File: Example/ExampleAppTests/PerformanceBDDTests.swift (hunk 1 above)
Comment (Vadimkomis) [reply Example/ExampleAppTests/PerformanceBDDTests.swift:24]: `identfier` should be `identifier`

Comment (bytedissident) [reply Example/ExampleAppTests/PerformanceBDDTests.swift]: 👍

File: Example/ExampleAppTests/PerformanceBDDTests.swift (hunk 2 above)
Comment (bytedissident) [reply Example/ExampleAppTests/PerformanceBDDTests.swift]: 👍

**Review by Vadimkomis — APPROVED** [review]
Very nice pr

**Pull request merged**